src/processcontroller.h

```diff
@@ -44,7 +44,7 @@ Q_OBJECT
             // if the process reports an error -- release the lock, report error, and return appropriate error code
             connect(process, &QProcess::errorOccurred, gui, [=]()
             {
-                gui->errorHandle(process->error());
+                gui->errorHandle(*process);
```

Comment:
I've changed the error handler to take the whole failed process invocation by const ref.

At the call-site this is a pointer, so it must be dereferenced to be passed to a const ref (sue me)

Reply:
Considering we use class pointers, it would be hypocritical of me to sue you for this.  Anyways, the approach seems fine, I just want more clarity on the changes.  Could you add some more comments in mainwindow.cpp to that effect?

Reply:
Comments like what?

Reply:
Elaborating on what line 118-125 do.  Just what/why is being printed.  So if the output is ever adjusted, people know which line corresponds to what.

Reply:
maybe later but now I think it's obvious, no?

Reply:
It can be done later, I may end up writing them myself, but I didn't want to misinterpret how you intend the code to work.